org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/markers/SonarMarker.java

```diff
@@ -97,76 +95,44 @@ private static Integer getPriority(final String severity) {
+      Integer startLine = issue.getStartLine();
+      if (startLine == null) {
+        return;
+      }
+      ITextFileBufferManager iTextFileBufferManager = FileBuffers.getTextFileBufferManager();
+      try {
+        iTextFileBufferManager.connect(iFile.getFullPath(), LocationKind.IFILE, new NullProgressMonitor());
+        ITextFileBuffer iTextFileBuffer = iTextFileBufferManager.getTextFileBuffer(iFile.getFullPath(), LocationKind.IFILE);
+        IDocument iDoc = iTextFileBuffer.getDocument();
+        int startLineStartOffset = iDoc.getLineOffset(startLine - 1);
+        Integer issueStartLineOffset = issue.getStartLineOffset();
+        if (issueStartLineOffset != null) {
+          markerAttributes.put(IMarker.CHAR_START, startLineStartOffset + issueStartLineOffset);
+          Integer issueEndLine = issue.getEndLine();
+          int endLineStartOffset = issueEndLine != startLine ? iDoc.getLineOffset(issueEndLine - 1) : startLineStartOffset;
+          markerAttributes.put(IMarker.CHAR_END, endLineStartOffset + issue.getEndLineOffset());
+        } else {
+          markerAttributes.put(IMarker.CHAR_START, startLineStartOffset);
+          int length = iDoc.getLineLength((int) startLine - 1);
+          String lineDelimiter = iDoc.getLineDelimiter(startLine - 1);
+          int lineDelimiterLength = lineDelimiter != null ? lineDelimiter.length() : 0;
+          markerAttributes.put(IMarker.CHAR_END, startLineStartOffset + length - lineDelimiterLength);
         }
       } catch (Exception e) {
         SonarLintCorePlugin.getDefault().error("Unable to compute position of SonarLint marker on resource " + resource.getName() + ": " + e.getMessage());
-      }
-    }
-  }
-
-  /**
-   * This class is used to count lines in a stream and keep char index. Contrary to {@link LineNumberReader} it preserve
-   * \r\n as 2 characters.
-   */
-  private static class LineAndCharCountReader extends BufferedReader {
-
-    /** The current line number */
-    private int lineNumber = 1;
-
-    private int charIndex = 0;
-
-    /** If the previous character was a cariage return */
-    private boolean cr = false;
-
-    public LineAndCharCountReader(Reader in) {
-      super(in);
-    }
-
-    public int getLineNumber() {
-      return lineNumber;
-    }
-
-    public int getCharIndex() {
-      return charIndex;
-    }
-
-    @Override
-    public int read() throws IOException {
-      int val = super.read();
-      if (val == -1) {
-        if (cr) {
-          lineNumber++;
-          cr = false;
-        }
-      } else {
-        charIndex++;
-        if (val == '\r') {
-          if (cr) {
-            lineNumber++;
-          } else {
-            cr = true;
-          }
-        } else if (val == '\n' || cr) {
-          lineNumber++;
-          cr = false;
+      } finally {
+        try {
+          iTextFileBufferManager.disconnect(iFile.getFullPath(), LocationKind.IFILE, new NullProgressMonitor());
+        } catch (CoreException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)